### PR TITLE
Replace hardcoded class with dunder class attribute

### PR DESCRIPTION
**Pull request by ueg1990**
Small suggestion as this was something I heard in Raymond Hettinger's talk at PyCon 2015

### CODE REVIEW

**Comment by ueg1990** [comment]
Btw if this is accepted, i can do it for the other modules/python files...in that way i can also read some of the requests code :stuck_out_tongue:

File: requests/structures.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ class LookupDict(dict):
 
     def __init__(self, name=None):
         self.name = name
-        super(LookupDict, self).__init__()
+        super(self.__class__, self).__init__()
```
Comment (jmoldow) [inline requests/structures.py:93]: This is not correct. The class name must always be hard-coded when using `super()`.

**Comment by Lukasa** [comment]
I certainly want the change in `super` removed: as pointed out in the diff comment by @jmoldow, that isn't valid. Otherwise, I suspect it makes the class somewhat more subclass-friendly, which I don't object to.

**Comment by ueg1990** [comment]
i am not sure why this is considered invalid. can you elaborate?

**Comment by jmoldow** [comment]
There is a reason that `super()` requires two arguments. It's because the first argument can't be deduced from `self`, because the first argument isn't (in general) `self.__class__`.

If you have a method-resolution order C -> B -> A -> object, `C.__init__` is going to call `super(C, self).__init__`, which will call `B.__init__`, which will call `super(B, self).__init__`, which will call `A.__init__`. The only call where the first argument is `self.__class__` is the first one, but you don't know in advance what will even be the first one.

The first argument needs to be the hard-coded class name, in order for super to figure out what the super object should refer to.

**Comment by sigmavirus24** [comment]
Actually `__class__` and `self.__class__` are a whole different level of magic that will thoroughly break this for any subclass. Sorry @ueg1990 this is not something I will allow in requests.

**Comment by ueg1990** [comment]
lol i do not have any objection to that. learnt something new in python :)
onto the next pull request

**Pull request closed without merging**